pkg/test/utils/kubernetes.go

```diff
@@ -1217,6 +1219,76 @@ func RunAssertCommands(ctx context.Context, logger Logger, namespace string, com
 	return RunCommands(ctx, logger, namespace, convertAssertCommand(commands, timeout), workdir, timeout, kubeconfigOverride)
 }
 
+// RunAssertExpressions evaluates a set of CEL expressions specified as AnyAllExpressions
```

Comment:
Let's not make this file longer than it needs to be:
- please keep the part that populates `variables` inside `(s *Step) CheckAssertExpressions()`
- the rest (that does the evaluation and result checking) should probably go into a new cel package; please also add some unit tests for this part